tools/release-posts/commands/release-post/release-post-release.ts

```diff
@@ -30,118 +36,179 @@ const program = new Command()
 	.description( 'CLI to automate generation of a release post.' )
 	.argument(
 		'<currentVersion>',
-		'The version of the plugin to generate a post for, please use the tag version from Github.'
-	)
-	.option( '--outputOnly', 'Only output the post, do not publish it' )
-	.option(
-		'--previousVersion <previousVersion>',
-		'If you would like to compare against a version other than last minor you can provide a tag version from Github.'
+		'The current version in x.y.z or x.y.z-stage.n format. Ex: 7.1.0, 7.1.0-rc.1'
 	)
-	.option(
-		'--tags <tags>',
-		'Comma separated list of tags to add to the post.',
-		'Releases,WooCommerce Core'
+	.argument(
+		'<previousVersion>',
+		'The previous version in x.y.z format. Ex: 7.0.0'
 	)
-	.action( async ( currentVersion, options ) => {
-		const tags = options.tags.split( ',' ).map( ( tag ) => tag.trim() );
+	.option( '--outputOnly', 'Only output the post, do not publish it' )
+	.option( '--editPost <postId>', 'Updates an existing post' )
+	.option( '--siteId <siteId>', 'For posting to a non-default site (for testing)')
+	.action( async ( currentVersion, previousVersion, options ) => {
 
-		const previousVersion = options.previousVersion
-			? semver.parse( options.previousVersion )
-			: semver.parse( currentVersion );
+		const siteId = options.siteId || DEVELOPER_WOOCOMMERCE_SITE_ID;
+		const tags = ( options.tags && options.tags.split( ',' ).map( ( tag ) => tag.trim() ) ) || [ 'WooCommerce Core','Releases' ];
+		const isOutputOnly = !! options.outputOnly;
+		const isEditPost = !! options.editPost;
```

Comment:
The way you're declaring this and then checking it before doing stuff with options.editPost, is either too smart for TS or not typesafe.

To get rid of all the warnings I would probably just remove this variable and then where you want to check for editPost just do:

`if (options.editPost) {` since an if statement in JS is falsy for: undefined, empty string etc. Perfectly fine for this case I think.

I also think probably this option should be named like editPostId or something?